### PR TITLE
[FLINK-18863][python] Support read_text_file() and print() interface …

### PR DESCRIPTION
**Pull request by shuiqiangchen**
## What is the purpose of the change

Add read_text_file() and print() interface for Python DataStream API. It would be of benefit for ease of use and debug purpose. Further more, Users can play with Flink out of box through these simple source/sink API.

## Brief change log

- Add print() for Python DataStream API.
- Add read_text_file for StreamExecutionEnvironment to get a DataStream.

## Verifying this change

This change has test cases covered by test_print() in test_data_stream.py and test_read_text_file() in test_stream_execution_environment.py

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-python/pyflink/datastream/data_stream.py [hunk 1]
```diff
@@ -483,7 +518,17 @@ def reduce(self, func: Union[Callable, ReduceFunction]) -> 'DataStream':
             j_python_data_stream_scalar_function_operator
         ))
 
-    def _values(self):
+    def key_by(self, key_selector: Union[Callable, KeySelector],
```
Comment (hequn8128) [inline flink-python/pyflink/datastream/data_stream.py:521]: The two methods are not supposed to be added in this PR?

File: flink-python/pyflink/datastream/stream_execution_environment.py [hunk 2]
```diff
@@ -465,6 +465,19 @@ def add_source(self, source_func: SourceFunction, source_name: str = 'Custom Sou
                                                                        j_type_info)
         return DataStream(j_data_stream=j_data_stream)
 
+    def read_text_file(self, file_path: str, charset_name: str = "UTF-8") -> DataStream:
```
[the remark this answers is not in the document]
Comment (shuiqiangchen) [reply flink-python/pyflink/datastream/stream_execution_environment.py:468]: Yes, it would be better to inform user it is not suitable for production.

**Comment by shuiqiangchen** [comment]
@hequn8128 Thank you for your comments, I have updated the PR, please have a look at it.

File: flink-python/pyflink/datastream/tests/test_stream_execution_environment.py [hunk 3]
```diff
@@ -254,3 +254,20 @@ def test_add_custom_source(self):
...
+                f.write('\n')
+
+        ds = self.env.read_text_file(text_file_path)
+        test_sink = DataStreamTestSinkFunction()
+        ds.add_sink(test_sink)
+        self.env.execute("test read text file")
+        results = test_sink.get_results()
+        results.sort()
+        texts.sort()
+        self.assertEqual(texts, results)
```
Comment (hequn8128) [inline flink-python/pyflink/datastream/tests/test_stream_execution_environment.py:273]: Maybe we can extract a assertEqualWithSort() method and reuse the method across all tests. We can improve this in a separate PR.

Comment (shuiqiangchen) [reply flink-python/pyflink/datastream/tests/test_stream_execution_environment.py:273]: Sure, I will do it in the later PR.

**Pull request merged**